modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj

```diff
@@ -982,3 +979,36 @@
                   qp.compile/compile-and-splice-parameters
                   :query
                   pretty-sql-lines))))))
+
+(deftest bigquery-process-stop-test
+  (mt/test-driver
+    :bigquery-cloud-sdk
+    (sync/sync-database! (mt/db) {:scan :schema})
+    (let [before-names (future-thread-names)]
+      (doseq [:let [callbacks (atom 0)]
+              [stop-tag stopper] [[:exception #(throw (Exception. "My Exception"))]
+                                  [:cancelled #(a/>!! qp.pipeline/*canceled-chan* true)]]
+              [tag callback] [[:initial-query stopper]
+                              [:during-page #(when (>= (swap! callbacks inc) 2)
+                                               (stopper))]]]
+        (testing (format "%s %s" tag stop-tag)
+          (reset! callbacks 0)
+          (binding [bigquery/*page-callback* callback
+                    bigquery/*page-size* 10]
+            (let [query  {:database (mt/id)
+                          :type "native"
+                          :native {:query (format "select * from `%s.orders` limit 100" test-db-name)}}
+                  result (mt/user-http-request :crowberto :post 202 "dataset" query)
+                  result (if (string? result)
+                           ;; If we get a string it means we streamed some partial results, the last line will be the
+                           ;; last streamed row + the full json exception.
+                           (-> result
+                               str/split-lines
+                               last
+                               (str/replace #"^[^\{]*\{" "{")
+                               (json/parse-string true))
+                           result)]
```

Comment:
This was not needed on master, I suspect the test libraries had updates to handle partial results better.